src/verse.zig

```diff
@@ -1,366 +1,49 @@
-    if (code == .no_content) {
-        @compileError("Sending JSON is not supported with status code no content");
-    }
-
-    vrs.status = code;
-    vrs.content_type = .{
-        .base = .{ .application = .json },
-        .parameter = .@"utf-8",
-    };
-
-    try vrs.quickStart();
-    const data = std.json.stringifyAlloc(vrs.alloc, json, .{
-        .emit_null_optional_fields = false,
-    }) catch |err| {
-        log.err("Error trying to print json {}", .{err});
-        return error.Unknown;
-    };
-    vrs.writeAll(data) catch |err| switch (err) {
-        error.BrokenPipe => |e| return e,
-        else => unreachable,
-    };
-}
-
-/// This function may be removed in the future
-pub fn quickStart(vrs: *Verse) NetworkError!void {
-    if (vrs.status == null) vrs.status = .ok;
-    switch (vrs.downstream) {
-        .http, .zwsgi => |_| {
-            vrs.sendHeaders() catch |err| switch (err) {
-                error.BrokenPipe => |e| return e,
-                else => unreachable,
-            };
-
-            vrs.writeAll("\r\n") catch |err| switch (err) {
-                error.BrokenPipe => |e| return e,
-                else => unreachable,
-            };
-        },
-        else => unreachable,
-    }
-}
-
-test "Verse" {
-    std.testing.refAllDecls(@This());
-}
-
-const std = @import("std");
-const Allocator = std.mem.Allocator;
-const eql = std.mem.eql;
-const Stream = std.net.Stream;
-const AnyWriter = std.io.AnyWriter;
-const bufPrint = std.fmt.bufPrint;
-const allocPrint = std.fmt.allocPrint;
-const splitScalar = std.mem.splitScalar;
-const log = std.log.scoped(.Verse);
-const iovec = std.posix.iovec;
-const iovec_c = std.posix.iovec_const;
+const frame = @import("frame.zig");
+const server = @import("server.zig");
+const router = @import("router.zig");
+const headers = @import("headers.zig");
+const request = @import("request.zig");
+const request_data = @import("request_data.zig");
+const cookies = @import("cookies.zig");
+const content_type = @import("content-type.zig");
+const template = @import("template.zig");
+const errors = @import("errors.zig");
+const static_file = @import("static-file.zig");
+const endpoint = @import("endpoint.zig");
+
+pub const Frame = frame.Frame;
+pub const RouteData = frame.RouteData;
+pub const Server = server.Server;
+pub const Router = router.Router;
+pub const Headers = headers.Headers;
+pub const Header = headers.Header;
+
+pub const Request = request.Request;
+pub const RequestData = request_data.RequestData;
+pub const QueryData = request_data.QueryData;
+pub const PostData = request_data.PostData;
+pub const DataKind = request_data.DataKind;
+pub const DataItem = request_data.DataItem;
+pub const Validator = request_data.Validator;
+
+pub const Cookie = cookies.Cookie;
+
+pub const ContentType = content_type.ContentType;
+
+pub const Template = template.Template;
+pub const PageData = template.PageData;
+pub const findTemplate = template.findTemplate;
```

Comment:
Don't use multiple lines when you only need to use one. 

The use of `pub const Name = @This();` when it a Type with fields is for convience inside the function. I'd rather drop `pub` from `pub const` than use these extra definitions;